src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -192,6 +192,8 @@ public Response addCluster(
       LOG.info("creating new cluster based on given seed host: {}", newCluster.getName());
       context.storage.addCluster(newCluster);
 
+      context.streamManager.addListeners(newCluster);
```

Comment:
This should not be necessary if the listeners are added in the JmxConnectionFactory, which would cover all cases of JMX connection.

Reply:
yes. and this only works if a cluster is registered first and in the same process. The reaper process can be restarted frequently and the stream manager listener will never be re-connected.